Create walkthrough blog post (#4381)
beetbox/beets: _posts/2022-06-21-walkthrough.md

---
title: "Beets: A Guided Walkthrough"
layout: main
section: blog
---

## Overview
This is a step by step walkthrough for getting started with the beets CLI using a sample music library. The sample music library includes public domain recordings of Debussy, Chopin, and Virginia Liston compiled from Open Music Archive, IMSLP, and Internet Archive.

We will use this library in its messy/unaltered state to showcase the power of the beets CLI by correcting obscure metadata and organizing the music library with the auto-tagger (and a little manual input). Lastly, we will host the library locally using the beets web plugin to stream the audio once finished and admire our results. Let's get started!

> If desired, there are resources included below to help find additional recordings that are in the public domain if you'd like to grow your sample music library. Note: be sure the release you're looking to include has metadata in the MusicBrainz database for best results.

Reviewer: Maybe it would be nice to link "MusicBrainz" to musicbrainz.org on the first mention?

Author: Can do!

## Installing and configuring beets

> Following are brief instructions for downloading and installing the beets CLI. For detailed instructions, head over to the official documentation here: [Getting Started](https://beets.readthedocs.io/en/stable/guides/main.html).

Reviewer: Just a somewhat more natural way to mark up the link:
```suggestion
Following are brief instructions for downloading and installing the beets CLI. For detailed instructions, head over to [the Getting Started guide](https://beets.readthedocs.io/en/stable/guides/main.html) in the official documentation.
```

```yaml
directory: ~/beets/music
library: ~/beets/data/musiclibrary.db
plugins: web
```

All I've done to support the above config is create the  `beets` directory inside my home folder, with empty child directories for `music` and `data` inside it. You can place your music library and data store wherever you like, and you don't need to create the `musiclibrary.db` file, it will be created for you.

## Importing the sample library
First, download the .zip archive of the sample music library from [this](https://bit.ly/3EZb5ue) public google drive file, extract the contents and take note of the path to the sample library for later use.

Take a second to scrub through the library's organization and inconsistent file naming. On a small scale this is a prime example of how messy a music library can be in its raw form, with no standard for metadata or organization.

Enter beets! We're going to clean up this mess and make our library beautiful.

Run `beet import {path to sample library}` to import the messy sample library and start using the beets auto-tagger.

In my case the path to the sample library is also in my home folder, so I will run the command `beet import ~/sample-library`

> The autotagger starts work on Album 1, and the closest match is selection `1`, go ahead and enter that in the prompt.

Reviewer: ```suggestion
The autotagger starts work on Album 1, and the closest match is selection `1`. Go ahead and enter that in the prompt.
```

Splitting some independent clauses again.

![](beets/_posts/2022-06-21-images/album1-find-tags.png)

Next, enter `a` to apply the changes found to Album 1.

![](beets/_posts/2022-06-21-images/album1-correct-tags.png)

Album 2 doesn't have a matching release, so we will enter one manually by entering `i` and then the following MusicBrainz id `96581fca-a7b1-4c94-bb23-7b0df9f00507`

![](beets/_posts/2022-06-21-images/album2-no-release.png)

Accept the changes for our manual match by selecting `a`

![](beets/_posts/2022-06-21-images/album2-correct-tags.png)

Album 3 doesn't have a matching release either, so we will enter one manually with `i` and then the following MusicBrainz id `95440be5-cff3-4222-8ddc-5b7ecb99a351`

![](beets/_posts/2022-06-21-images/album3-no-release.png)

> We've located the correct album, however we only have 11/48 tracks across a compilation of 3 CDs, hence the large output warning us of missing tracks.

Reviewer: ```suggestion
We've located the correct album; however, we only have 11/48 tracks across a compilation of 3 CDs, hence the large output warning us of missing tracks.
```

Fix usage of "however" (which is [not a conjunction](https://capra.cs.cornell.edu/styleguide/#however)).

For now, accept the changes for our manual match by selecting `a`

![](beets/_posts/2022-06-21-images/album3-apply.png)

Album 4 is a series of singletons with a number of matches found by the autotagger.

Accept these changes and import the tracks with the `t` flag for individual tracks

![](beets/_posts/2022-06-21-images/album4-find-tags.png)

The output should look like the following:

![](beets/_posts/2022-06-21-images/album4-import-tracks.png)

Finally, let's check out our library in its current state with `beet list`

![](beets/_posts/2022-06-21-images/beets-list.png)

We've made significant progress! Now let's correct some discrepencies that came up between the autotagger and our library.

## Correcting a mistake in the sample library
We can use the command `beet fields` to see all the metadata that can be updated for each track in the library, in our case we're interested in updating the `title` and `mb_trackid` fields to correct a series of mistakes in a couple of our albums.

### Mistake: Album 2 - Swapped track titles and metadata

> There are two tracks in our download of Moravec's Debussy album where the titles and metadata are swapped. "Children's Corner: Jimbo's Lullaby" and "Clair de lune" incorrectly refer to the opposite recording, let's change that with beets!

Reviewer: ```suggestion
There are two tracks in our download of Moravec's Debussy album where the titles and metadata are swapped. "Children's Corner: Jimbo's Lullaby" and "Clair de lune" incorrectly refer to the opposite recording. Let's change that with beets!
```
